src/observer.ts

```diff
@@ -12,10 +12,29 @@ export function observer<P extends object, TRef = {}>(
 ): React.MemoExoticComponent<
     React.ForwardRefExoticComponent<React.PropsWithoutRef<P> & React.RefAttributes<TRef>>
 >
+
 export function observer<P extends object>(
     baseComponent: React.FunctionComponent<P>,
     options?: IObserverOptions
 ): React.FunctionComponent<P>
+
+export function observer<
+    C extends React.FunctionComponent<any> | React.RefForwardingComponent<any>,
```

Comment:
I am probably missing something here, but why can't we have `P extends object` here as well and then...

```suggestion
    C extends React.FunctionComponent<P> | React.RefForwardingComponent<unknown>,
```

Not sure about `unknown` if it can cause some issue, but sounds better than `any` imo.

Reply:
the reason we have `any` here is that we want to prevent TypeScript to 'guess' a P (which often ends up in an `any` if it can't if there are generics etc), rather we compute it at the end (`infer P`) at the return position, based on the component type that was passed in. This approach is in my experience much more robust